### PR TITLE
🐛 Destination Redshift: Improved discovery for redshift-destination not SUPER streams

### LINKED ISSUE #12843: Destination Redshift fails after upgrade
## Environment
- **Airbyte version**:  example is 0.38.2-alpha
- **OS Version / Instance**: 
- **Deployment**: Kubernetes
- **Source Connector and version**: any 
- **Destination Connector and version**: Redshift 0.3.33  
- **Severity**: Medium
- **Step where error happened**: Sync job

## Current Behavior
I upgraded airbyte from 0.35 to 0.38, which also upgraded connectors. Redshift destination went from 0.3.23 to 0.3.32 (I also manually tried 0.3.33 but the problem persisted)

I found out that now redshift destination forces all `_airbyte_raw_*` tables to use the `SUPER` type instead of `VARCHAR`, but that causes the destination to fail on close with the exceptions below.

I also noticed that the relevant PRs say that is an option, but I can't find anything to disable it and return to the old behaviour.

## Expected Behavior
syncs to Redshift still work after upgrade

## Logs
*If applicable, please upload the logs from the failing operation. 
For sync jobs, you can download the full logs from the UI by going to the sync attempt page and 
clicking the download logs button at the top right of the logs display window.*

<details>
<summary>LOG</summary> 

[logs-6007.txt](https://github.com/airbytehq/airbyte/files/8688512/logs-6007.txt)
[logs-6008.txt](https://github.com/airbytehq/airbyte/files/8688514/logs-6008.txt)

</details>

## Steps to Reproduce
1. sync with old redshift versions. I have multiple connections in the same schema (with different prefixes)
2. upgrade redshift
3. sync again

### PR DESCRIPTION
**Pull request by alexandertsukanov**
## What
In the scope of the [bug](https://github.com/airbytehq/airbyte/issues/12843) was found, that the Redshift connector, could affect **not related** tables and VIEWS, during SUPER migration (By the not related, I mean the streams were not selected on UI).

## How
In the scope of this PR, discovery behavior was changed in a next way:

- Now the discovery searches only for tables (Streams), NOT VIEWS.
- It searches only selected tables (Streams) on UI.

## Recommended reading order
1. `x.java`

## 🚨 User Impact 🚨
During SUPER migration the VIEWS that depend on VARCHAR `_airbyte_data` columns could be DROPPED during migration. **So, they should be exported BEFORE the migration.**

## Pre-merge Checklist
Expand the relevant checklist and delete the others.

<details><summary><strong>New Connector</strong></summary>

### Community member or Airbyter

- [ ] **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret`
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [ ] `docs/integrations/README.md`
    - [ ] `airbyte-integrations/builds.md`
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
- [ ] After the connector is published, connector added to connector index as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
- [ ] Seed specs have been re-generated by building the platform and committing the changes to the seed spec files, as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)

</details>

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- [ ] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret`
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub and connector version bumped by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)

</details>

<details><summary><strong>Connector Generator</strong></summary>

- [ ] Issue acceptance criteria met
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] If adding a new generator, add it to the [list of scaffold modules being tested](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/generator/build.gradle#L41)
- [ ] The generator test modules (all connectors with `-scaffold` in their name) have been updated with the latest scaffold by running `./gradlew :airbyte-integrations:connector-templates:generator:testScaffoldTemplates` then checking in your changes
- [ ] Documentation which references the generator is updated as needed

</details>

## Tests

<details><summary><strong>Unit</strong></summary>

*Put your unit tests output here.*

</details>

<details><summary><strong>Integration</strong></summary>

*Put your integration tests output here.*

</details>

<details><summary><strong>Acceptance</strong></summary>

*Put your acceptance tests output here.*

</details>

### CODE REVIEW

**Review by VitaliiMaltsev — APPROVED** [review]

File: airbyte-integrations/connectors/destination-redshift/src/main/java/io/airbyte/integrations/destination/redshift/operations/RedshiftSqlOperations.java [hunk 1]
```diff
@@ -119,33 +127,61 @@ public boolean isValidData(final JsonNode data) {
...
+
+  /**
+   * @param database   - Database object for interacting with a JDBC connection.
    * @param schemaName - schema to update.
+   * @param tablesName - tables to update.
    */
-  private List<String> discoverNotSuperTables(final JdbcDatabase database,
-                                              final String schemaName) {
-    List<String> schemaAndTableWithNotSuperType = new ArrayList<>();
+  private List<String> discoverNotSuperTables(final JdbcDatabase database, final String schemaName, final String tablesName) {
```
Comment (tuliren) [inline airbyte-integrations/connectors/destination-redshift/src/main/java/io/airbyte/integrations/destination/redshift/operations/RedshiftSqlOperations.java:181]: Nitpick.

```suggestion
  private List<String> discoverNotSuperTables(final JdbcDatabase database, final String schemaName, final String tableName) {
```

**Review by tuliren — APPROVED** [review]
Looks good. I have two comments:
- If it is not too much extra work, can you add a test case in the existing unit test or integration test? Like adding an extra table in Redshift that won't be included in the catalog, and assure that it is not affected at the end.
- Can you run code formatter to fix some of the formatting? Right now we don't have module level Java formatting command. But you can run `./gradlew :spotlessJavaApply`, which will format all Java files. You can then just check in the files relevant to this PR, and ignore the rest of the formatted files.

**Comment by alexandertsukanov** [comment]
/publish connector=connectors/destination-redshift run-tests=false

**Pull request merged**